span-normalizer/src/test/java/org/hypertrace/core/spannormalizer/fieldgenerators/HttpFieldsGeneratorTest.java

```diff
@@ -567,32 +567,20 @@ public void testResponseStatusCodeTagKeysPriority() {
   }
 
   @Test
-  public void testInvalidUrl() {
+  public void testIncompleteUrl() {
     HttpFieldsGenerator httpFieldsGenerator = new HttpFieldsGenerator();
 
-    Map<String, JaegerSpanInternalModel.KeyValue> tagsMap1 = new HashMap<>();
-    tagsMap1.put(RawSpanConstants.getValue(HTTP_URL), createKeyValue("/dispatch/test?a=b&k1=v1"));
+    Map<String, JaegerSpanInternalModel.KeyValue> tagsMap = new HashMap<>();
+    tagsMap.put(RawSpanConstants.getValue(HTTP_URL), createKeyValue("/dispatch/test?a=b&k1=v1"));
 
-    Event.Builder eventBuilder1 = Event.newBuilder();
-    Http.Builder httpBuilder1 = httpFieldsGenerator.getProtocolBuilder(eventBuilder1);
+    Event.Builder eventBuilder = Event.newBuilder();
+    Http.Builder httpBuilder = httpFieldsGenerator.getProtocolBuilder(eventBuilder);
 
-    tagsMap1.forEach(
+    tagsMap.forEach(
         (key, keyValue) ->
-            httpFieldsGenerator.addValueToBuilder(key, keyValue, eventBuilder1, tagsMap1));
-
-    Assertions.assertNull(httpBuilder1.getRequestBuilder().getUrl());
-
-    Map<String, JaegerSpanInternalModel.KeyValue> tagsMap2 = new HashMap<>();
-    tagsMap2.put(RawSpanConstants.getValue(HTTP_URL), createKeyValue("http://abc.xyz/dispatch/test?a=b&k1=v1"));
```

Comment:
So this is not a valid URL anymore? Also how does this affect ML services that expect the full URL and not just the path when they read getUrl()?

Reply:
Both the URLs("http://abc.xyz/dispatch/test?a=b&k1=v1" and "/dispatch/test?a=b&k1=v1") are now considered valid(with this change), whereas previously we weren't considering the second one as a valid URL.

Not exactly sure about how it is going to affect the ML services but they need to be aware that they may not get a full URL, after this change. If required, they may have to check on their end if it is a full URL or not. Need to see if those services will require any change.